### ISSUE TITLE
TestSdsVaultCaFlow is constantly failing for the past week

### ISSUE THREAD

**Issue by yangminzhu** [istio/istio#15230]
(NOTE: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities/>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>
)

**Bug description**
It started to fail in post-submit since 06/23.
https://k8s-testgrid.appspot.com/istio-postsubmits-master#integ-security-k8s-postsubmit-tests&width=20

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by lei-tang** [comment]
This problem may result from the pod running the testing Vault server is restarted by GKE, which necessitates initialization, unseal, and configuration of the testing Vault server. I have initialized, unsealed, and configured the testing Vault server.
This test is in postsubmit already and  I agree with John's point in https://github.com/istio/istio/pull/15233 on this test failure: "I think it's fine to keep it failing in postsubmit so you can stop least see it's failing when you look at the post submits".
I will monitor the status of this test in postsubmits.

**Comment by yangminzhu** [comment]
I don't know how is the Vault server started but I think the test has recovered now, @lei-tang is it possible to setup and start the Vault server together with the test? It could be a docker image and deployed inside the cluster for testing, so that we don't need to depend on an external Vault server in a separate project and causes test failure like this.

**Comment by lei-tang** [comment]
There are some manual steps when configuring Vault server, e.g., unseal, login, etc, which makes it difficult to start the Vault server with the test. But I will investigate how the Vault server can be improved, e.g., docker image, k8s auto recovery, etc.

**Comment by yangminzhu** [comment]
@lei-tang It seems the test started to fail again.

**Comment by lei-tang** [comment]
Changed the Vault pod with restartPolicy=never. TestSdsVaultCaFlow passed at https://prow.k8s.io/view/gcs/istio-prow/logs/integ-security-k8s-postsubmit-tests-master/282. I will continue monitoring this test.

**Comment by howardjohn** [comment]
@lei-tang where was the PR for that change?

**Comment by lei-tang** [comment]
The change of the Vault pod with restartPolicy=never is on the GKE cluster hosting the Vault server and is not in the Istio code base (hence no PR). It is uncertain that the pod with restartPolicy=never will eliminate the problem because if the GKE node restarts, the Vault server still needs to be manually unsealed and logged in again.

**Comment by howardjohn** [comment]
I don't think we should have tests depending on a cluster that you are manually changing values around. If we HAVE to have this magic vault cluster can the config at least be in the istio repo so it can be understood by others and be reproducible. I really don't think we should have a test that is going to break when a node restarts -- node restarts are not rare

**Comment by lei-tang** [comment]
@howardjohn How about disable this test until we have a solution for node restarting problem?
